src/spreadsheets.js

```diff
@@ -0,0 +1,286 @@
+    modelListeners: {
+        '*': {
+            func: 'spreadsheets.spreadsheet.setRemoteContent',
+            args: ['{that}', '{apiClient}', '{change}.path.0', '{change}.value'],
+            // FIXME: currently, this model relay doesn't work as intended. 
+            // It should systematically exclude mutations coming from reads. 
+            // Can those be associated with a particular invoker on the apiClient sub-component or locally? 
+            // The kludge I can make right now is to check the written value against the last-read value.
+            // excludeSource: 'init' does not seem to be what I want, because multiple reads could happen over the component life cycle.
+        }
+    }
+});
+
+// TODO: refactor this messy function
+// TODO: I wonder if I can more declaratively specify
+//   which api call with which body
+//   how to decode and extract data
+//   how to fill the model
+// and whether that has value for the design
+spreadsheets.spreadsheet.getRemoteContent = function(that, apiClient) {
+    var metadataPromise = apiClient.getSpreadsheet(apiClient, that.options.spreadsheetId);
+    var requests = [];
+    metadataPromise.then(function(res) {
+        fluid.each(res.data.sheets, function(sheetObject) {
+            var {title} = sheetObject.properties;
+            var {columnCount, rowCount} = sheetObject.properties.gridProperties;
+            requests.push({title, columnCount, rowCount});
+        });
+        var ranges = [];
+        fluid.each(requests, function({title, columnCount, rowCount}) {
+            var boundingRange = title + '!' + 'A1:' + spreadsheets.coordToA1(columnCount, rowCount);
+            that.sheetBounds[title] = boundingRange;
+            ranges.push(boundingRange);
+        });
+        var valuePromise = apiClient.getRanges(
+            apiClient,
+            that.options.spreadsheetId,
+            ranges);
+        valuePromise.then(function(res) {
+            fluid.each(res.data.valueRanges, function({range, values}) {
+                var sheetName = range.split('!')[0];
+                that.applier.change(sheetName, values);
+            });
+        });
+    }, function(error) {
+        fluid.log(fluid.logLevel.WARN, error);
+    });
+};
+
+/**
+ * Convert a column and row to A1 notation, the conventional coordinate system of spreadsheets.
+ * @param {Number} column The 1-indexed column number.
+ * @param {Number} row The 1-indexed row number.
+ * @returns {String} The coordinate in A1 notation.
+ */
+spreadsheets.coordToA1 = function(column, row) {
+    return numberToColumn(column) + row;
+};
+
+var numberToColumn = function(int) {
+    var togo = int - 1;
+    togo = togo.toString(26);
+    togo = successorBase27(togo);
+    togo = shiftToAlphabetical(togo);
+    return togo;
+};
+
+var successorBase27 = function(str) {
+    var togo;
+    for (i = str.length - 1; i >= 0; i--) {
+        var char = str[i];
+        if (char !== 'q') {
```

Comment:
Some explanation of the strategy here would be helpful. I think you yourself would also appreciate some test cases for this algo and also split it off into its own file.